openedx_ledger/api.py

```diff
@@ -147,3 +153,40 @@ def create_ledger(unit=None, idempotency_key=None, subsidy_uuid=None, initial_de
         )
 
     return ledger
+
+
+def create_adjustment(ledger, quantity, reason, notes, transaction_of_interest=None, **metadata):
+    """
+    Creates a new Transaction and related Adjustment record
+    to adjust the balance of the given ledger.
+    """
+    breakpoint()
+    idempotency_key = f'{ledger.uuid}-adjustment-{quantity}-reason-{uuid.uuid4()}'
+    transaction = create_transaction(
+        ledger,
+        quantity,
+        idempotency_key=idempotency_key,
+        **metadata,
+    )
+    try:
+        adjustment = models.Adjustment.objects.create(
+            ledger=ledger,
+            quantity=quantity,
+            idempotency_key=idempotency_key,
+            transaction=transaction,
+            transaction_of_interest=transaction_of_interest,
+            reason=reason,
+            notes=notes,
+        )
+        transaction.state = models.TransactionStateChoices.COMMITTED
+        adjustment.state = models.TransactionStateChoices.COMMITTED
```

Comment:
[Question] is this not settable in the create statement?

Reply:
Ah, yeah it is.  I'm cleaning up this whole block (based also on your comment below).